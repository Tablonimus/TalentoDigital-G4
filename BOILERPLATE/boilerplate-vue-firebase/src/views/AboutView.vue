<template>
  <div class="default-layout">
    <Navbar
      title="Vista Contacto"
      logo="https://st2.depositphotos.com/1768926/10934/v/450/depositphotos_109347560-stock-illustration-water-wave-logo-template.jpg"
      :showLinks="true"
    />
    <!--  -->
    <main class="h-100 ">
      <h1>This is an about page</h1>
    </main>
    <!--  -->
    <FooterComponent
      title="Gracias por elegirnos"
      logo="https://st2.depositphotos.com/1768926/10934/v/450/depositphotos_109347560-stock-illustration-water-wave-logo-template.jpg"
    />
  </div>
</template>

<script>
import Navbar from "../components/NavbarComponent.vue";
import FooterComponent from "../components/FooterComponent.vue";

export default {
  name: "AboutView",
  components: {
    Navbar,
    FooterComponent,
  },
};
</script>
