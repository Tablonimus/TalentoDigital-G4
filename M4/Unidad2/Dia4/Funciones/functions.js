/* FUNCIONES */
/*
function nombre(...parametros){
 <código a ejecutar>
}
*/

/* DECLARACIÓN DE UNA FUNCIÓN */
function suma(numero1, numero2) {
  let resultado = numero1 + numero2;

  console.log(resultado);
}

/* INVOCACIÓN DE UNA FUNCIÓN */
suma(10, 22);
