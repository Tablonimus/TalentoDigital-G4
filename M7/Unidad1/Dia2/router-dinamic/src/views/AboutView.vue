<template>
  <div class="about">
    <h1>This is an about page</h1>
    <router-link to="/services">
      <button>Ir a servicios</button>
    </router-link>
  </div>
</template>
