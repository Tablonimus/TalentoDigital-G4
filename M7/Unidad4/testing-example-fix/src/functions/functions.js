function sumarNumeroPor10(numero) {
  return numero + 10;
}

export { sumarNumeroPor10 };
