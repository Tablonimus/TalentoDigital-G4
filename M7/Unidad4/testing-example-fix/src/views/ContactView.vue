<template>
    <div class="home">Contacto</div>
  </template>
  
  <script>
  export default {
    name: "ContactView",
  };
  </script>
  