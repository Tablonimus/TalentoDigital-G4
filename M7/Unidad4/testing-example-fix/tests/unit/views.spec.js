import { mount } from "@vue/test-utils";
import ContactView from "@/views/ContactView.vue";
import AboutView from "@/views/AboutView.vue";
import { sumarNumeroPor10 } from "@/functions/functions";
import { createRouter, createWebHistory } from "vue-router";

describe("Tests de vista About", () => {
  it("Test1 - Matchea el snapshot de la vista About Correctamente", () => {
    /* Seleccionar el componente que será puesto a prueba */
    const wrapper = mount(AboutView);
    /* Assertion => Aserción */
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("Test2 - Probar la existencia del componente en la ruta", async () => {
    const routes = [
      {
        path: "/about",
        name: "about",
        component: () => AboutView,
      },
    ];
    const router = createRouter({
      history: createWebHistory(process.env.BASE_URL),
      routes,
    });

    router.push("/about");
    await router.isReady();

    const wrapper = mount(AboutView, {
      global: {
        plugins: [router],
      },
    });

    expect(wrapper.findComponent(AboutView).exists()).toBe(true);
  });
});

describe("Tests de vista Contact", () => {
  it("Test1 - Matchea el snapshot de la vista Contact Correctamente", () => {
    /* Seleccionar el componente que será puesto a prueba */
    const wrapper = mount(ContactView);
    /* Assertion => Aserción */
    expect(wrapper.html()).toMatchSnapshot();
  });

});

xdescribe("Test de constante de número", () => {
  const numero = 5;

  it("Numero no puede ser menor a 5", () => {
    expect(numero < 5).toBe(false);
  });

  it("Numero no puede ser mayor a 5", () => {
    expect(numero > 5).toBe(false);
  });

  it("Numero es igual a 5", () => {
    expect(numero === 5).toBe(true);
  });

  it("Numero es impar", () => {
    expect(numero % 2 !== 0).toBe(true);
  });

  it("Numero + 10  da como resultado 15", () => {
    const resultado = numero + 10;
    expect(resultado).toEqual(15);
  });
});

xdescribe("Test de funcion sumarPor10", () => {
  it("suma correctamente el número por 10", () => {
    const numero = 78;

    expect(sumarNumeroPor10(numero)).toEqual(numero + 10);
  });
});
