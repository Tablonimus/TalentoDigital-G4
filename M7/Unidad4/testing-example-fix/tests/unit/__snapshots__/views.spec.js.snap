// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Suite de vista About Test1 - Matchea el snapshot de la vista About Correctamente 1`] = `
<div class="about">
  <h1>This is an about page</h1>
</div>
`;

exports[`Test Suite de vista Contact Test1 - Matchea el snapshot de la vista Contact Correctamente 1`] = `<div class="home">Contacto</div>`;

exports[`Test de vista About Match de snapshot con AboutView.vue 1`] = `
<div class="about">
  <h1>This is an about page</h1>
</div>
`;

exports[`Tests de vista About Test1 - Matchea el snapshot de la vista About Correctamente 1`] = `
<div class="about">
  <h1>This is an about page</h1>
</div>
`;

exports[`Tests de vista Contact Test1 - Matchea el snapshot de la vista Contact Correctamente 1`] = `<div class="home">Contacto</div>`;
